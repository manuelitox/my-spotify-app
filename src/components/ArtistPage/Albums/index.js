import React from 'react'

import Title from 'components/ArtistPage/Title/'
import Album from './Album/'
import Button from 'components/generic/Button/'
import StyledWrapperAlbums from './styles'

const Albums = () => (
  <div>
    <Title>Albums</Title>
    <StyledWrapperAlbums>
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
    </StyledWrapperAlbums>
    <Button onClick={ () => console.log('open albums list popup') }>view more albums</Button>
  </div>
)

export default Albums