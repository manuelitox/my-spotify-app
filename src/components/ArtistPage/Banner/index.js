import React from 'react'

import StyledBanner from './styles'
import BannerWrapper from './wrapper'
import StylesWaveArtistIcon from './stylesWaiveIcon'

const Banner = () => (
  <StyledBanner>
    <BannerWrapper />
    <StylesWaveArtistIcon />
  </StyledBanner>
)

export default Banner