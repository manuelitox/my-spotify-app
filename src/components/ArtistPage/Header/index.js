import React from 'react'
import { Link } from 'react-router-dom'

import StyledHeader from './styles'
import SearchFormContainer from 'components/generic/SearchForm/container'
import { ReactComponent as HeadsetIcon } from 'svgs/headset.svg'

const Header = () => (
  <StyledHeader>
    <Link to="/">
      <HeadsetIcon height="45" width="55" viewBox="0 0 70 55"/>
    </Link>
    <SearchFormContainer placeholder="search another artist..." />
  </StyledHeader>
)

export default Header