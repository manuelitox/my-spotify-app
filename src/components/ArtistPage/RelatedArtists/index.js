import React, { Component } from 'react'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Button from 'components/generic/Button/'
import Artist from './Artist/'
import StyledRelatedArtists from './styles'
import StyledWrapperRelatedArtists from './styles-wrapper'
import StyledContainerRelatedArtists from './styles-container'

class RelatedArtists extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    return (
      <StyledRelatedArtists>
        <StyledContainerRelatedArtists>
          <Title color={ theme.blueDark }>Related Artists</Title>
          <StyledWrapperRelatedArtists>
            <Artist id="1" name="Martin Abasto" />
            <Artist id="2" name="Bairam Frootan" />
            <Artist id="3" name="Emmalynn Mazzia" />
            <Artist id="4" name="Gabriel Pires" />
            <Artist id="5" name="Irene Sotelo" />
            <Artist id="6" name="Oea Romana" />
            <Artist id="7" name="Pin Jung-Eym" />
            <Artist id="8" name="Udom Paowsong" />    
            <Artist id="9" name="John Doe" />    
          </StyledWrapperRelatedArtists>
          <Button 
            bg={ theme.blueDark }
            textColor="white"
            onClick={ this.togglePopUp }>
            view related artists
          </Button>
          <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
            <div>
              <h1>Related Artists list</h1>
              <p>description...</p>
            </div>
          </PopUp>          
        </StyledContainerRelatedArtists>
      </StyledRelatedArtists>
    )
  }
}

export default RelatedArtists