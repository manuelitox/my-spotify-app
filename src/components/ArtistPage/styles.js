import styled from 'styled-components'

const StyledArtistPage = styled.div`
  background-color: ${ props => props.theme.blueDark };
`

export default StyledArtistPage