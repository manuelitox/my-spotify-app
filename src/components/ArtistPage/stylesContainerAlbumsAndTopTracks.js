import styled from 'styled-components'

import WithContainer from 'components/generic/Container/'

const StyledContainerAlbumsAndTopTracks = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  padding-bottom: 40px;
  padding-top: 40px;
  @media only screen and (max-width: ${ props => props.theme.size.medium }) {
    grid-template-columns: 1fr;
    & > div:first-of-type {
      margin-bottom: 30px;
    }
  }
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    padding-top: 0;
  }  
`

export default WithContainer(StyledContainerAlbumsAndTopTracks)