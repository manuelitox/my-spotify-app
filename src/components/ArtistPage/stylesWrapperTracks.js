import styled from 'styled-components'

const StylesWrapperTracks = styled.div`
  & > div {
    margin: 20px 0;
  }
`

export default StylesWrapperTracks