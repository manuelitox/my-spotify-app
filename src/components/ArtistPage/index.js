import React from 'react'
 
import Header from './Header/'
import Banner from './Banner/'
import Albums from './Albums/'
import StyledArtistPage from './styles'
import StyledContainerAlbumsAndTopTracks from './stylesContainerAlbumsAndTopTracks'

const ArtistPage = () => (
  <StyledArtistPage>
    <Header />
    <Banner />
    <StyledContainerAlbumsAndTopTracks>
      <Albums />
      <div>tracks</div>
    </StyledContainerAlbumsAndTopTracks>
  </StyledArtistPage>
)

export default ArtistPage