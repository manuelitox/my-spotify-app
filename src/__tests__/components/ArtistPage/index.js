import React from 'react'
import { shallow } from 'enzyme'

import { ArtistPage } from 'components/ArtistPage/'

describe('Component | ArtistPage', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <ArtistPage getArtist={ jest.fn() } />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})