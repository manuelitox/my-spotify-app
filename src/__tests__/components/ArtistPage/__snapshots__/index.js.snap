// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | ArtistPage should render a component 1`] = `
<styled.div>
  <Header />
  <Banner />
  <Component>
    <Albums />
    <div>
      tracks
    </div>
  </Component>
</styled.div>
`;
